src/main/java/com/ss/training/utopia/agent/controller/AgentController.java

```diff
@@ -8,32 +8,77 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
 /**
  * @author Trevor Huis in 't Veld
  */
 @RestController
-@RequestMapping(path="/agent")
+@RequestMapping(path = "/agent")
 public class AgentController {
 
-    @Autowired
+	@Autowired
 	AgentService service;
 
-	@GetMapping(path="/bookings/{agentId}")
-    public ResponseEntity<Booking[]> getAllBookingsByAgent(@PathVariable long agentId) {
+	@PostMapping(path = "/booking")
+	public ResponseEntity<Booking> createBooking(@RequestBody Booking booking) {
+		HttpStatus status = HttpStatus.BAD_REQUEST;
+
+		String bookingResult = service.createBooking(booking);
+		switch (bookingResult) {
+			case("Card Declined"):
+				break;
+			case("Flight Full"):
+				status = HttpStatus.NO_CONTENT;
```

Comment:
If you're going to return the no content status, it's probably also best to send null in the body instead of a booking object. Or maybe a different status would be more fitting.

Reply:
I had a lot of trouble picking this status out. I'll add the null content.